packages/integrations/mdx/src/vite-plugin-mdx-postprocess.ts

```diff
@@ -37,9 +37,9 @@ export function vitePluginMdxPostprocess(astroConfig: AstroConfig): Plugin {
 /**
  * Inject `Fragment` identifier import if not already present.
  */
-function injectFragmentImport(code: string, imports: readonly ImportSpecifier[]) {
-	if (!isSpecifierImported(code, imports, fragmentImportRegex, 'astro/jsx-runtime')) {
-		code += `\nimport { Fragment } from 'astro/jsx-runtime';`;
+function injectUnderscoreFragmentImport(code: string, imports: readonly ImportSpecifier[]) {
+	if (!isSpecifierImported(code, imports, underscoreFragmentImportRegex, 'astro/jsx-runtime')) {
```

Comment:
I changed the `Fragment` detection to `_Fragment` as it's often already generated by MDX, which should reduce the code generation.